app/views/users/edit.html.erb

```diff
@@ -0,0 +1,21 @@
+<%= form_for @user do |f| %>
+    <%=f.label :username, "username:" %>
+    <%=f.text_field :username%>
+    <%=f.label :email, "email:" %>
+    <%=f.email_field :email%>
+    <%=f.label :password, "password:" %>
+    <%=f.password_field :password%>
+    <%=f.submit "Update" %>
+<%end%>
+
+<% if @user.errors.any? %>
+    <div id="error_explanation">
+      <h2><%= pluralize(@user.errors.count, "error") %> prohibited this user from being saved:</h2>
+
+      <ul>
+      <% @user.errors.full_messages.each do |msg| %>
+        <li><%= msg %></li>
+      <% end %>
+      </ul>
+    </div>
+  <% end %>
```

Comment:
* [ ] There should be absolutely NO logic in views.

  > Views are only for displaying information that became available to it from its corresponding controller not generating it. 

- check this guide * [The Beginner's Guide to Rails Helpers](https://mixandgo.com/learn/the-beginners-guide-to-rails-helpers) that shows how to put if/else statements in a helper method instead of the view [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/section/no_logic_in_helpers_requirement.png) from Pathwright

- What I wanted you to do here is to replace lines 32 to 42 (the if statement) for a method call, such method should be placed in your helpers/UserHelper module and it should handle the logic within this if statement because there should be no logic placed in your views

Reply:
Thanks for your suggestions. changes made